### ISSUE TITLE
virtualbox-vm unknown configuration key: "guest_additions_interface" and "guest_os_type"

### ISSUE THREAD

**Issue by harisund** [hashicorp/packer#9180]
#### Overview of the Issue

I just created a VM, installed Ubuntu on it and created a template. Now I am trying to call packer against that template, hence the use of "virtualbox-vm" builder.

### Packer version

1.5.6

### Simplified Packer Buildfile

This is the output of `packer fix <my-file>.json`

```
{
  "builders": [
    {
      "acpi_shutdown": "true",
      "attach_snapshot": "base-install",
      "guest_additions_interface": "sata",
      "guest_additions_mode": "attach",
      "guest_os_type": "Ubuntu_64",
      "headless": "false",
      "keep_registered": "true",
      "skip_export": "true",
      "ssh_password": "password",
      "ssh_timeout": "3m",
      "ssh_username": "ubuntu",
      "target_snapshot": "built",
      "type": "virtualbox-vm",
      "vm_name": "hjb"
    }
  ],
  "provisioners": [
    {
      "inline": "whoami",
      "type": "shell"
    }
  ]
}
```

### Operating system and Environment details

Windows 10
VirtualBox 6.0.12
Packer 1.5.6

### Log Fragments and crash.log files

```
Template validation failed. Errors are shown below.

Errors validating build 'virtualbox-vm'. 2 errors occurred:
        * unknown configuration key: "guest_additions_interface"; raws is []interface {}{map[string]interface {}{"acpi_shutdown":"true", "attach_snapshot":"base-install", "guest_additions_interface":"sata", "guest_additions_mode":"attach", "guest_os_type":"Ubuntu_64", "headless":"false", "keep_registered":"true", "skip_export":"true", "ssh_password":"password", "ssh_timeout":"3m", "ssh_username":"ubuntu", "target_snapshot":"built", "vm_name":"hjb"}, map[string]interface {}{"packer_build_name":"virtualbox-vm", "packer_builder_type":"virtualbox-vm", "packer_debug":false, "packer_force":false, "packer_on_error":"", "packer_template_path":"C:\\harisund\\code\\packer\\ubuntu-vbox-vm-no-vars.json", "packer_user_variables":map[interface {}]interface {}{}}}

 and ctx data is map[interface {}]interface {}(nil)
        * unknown configuration key: "guest_os_type"; raws is []interface {}{map[string]interface {}{"acpi_shutdown":"true", "attach_snapshot":"base-install", "guest_additions_interface":"sata", "guest_additions_mode":"attach", "guest_os_type":"Ubuntu_64", "headless":"false", "keep_registered":"true", "skip_export":"true", "ssh_password":"password", "ssh_timeout":"3m", "ssh_username":"ubuntu", "target_snapshot":"built", "vm_name":"hjb"}, map[string]interface {}{"packer_build_name":"virtualbox-vm", "packer_builder_type":"virtualbox-vm", "packer_debug":false, "packer_force":false, "packer_on_error":"", "packer_template_path":"C:\\harisund\\code\\packer\\ubuntu-vbox-vm-no-vars.json", "packer_user_variables":map[interface {}]interface {}{}}}

 and ctx data is map[interface {}]interface {}(nil)

```

**Comment by nywilken** [comment]
Hi there, thanks for reaching out. This looks like a documentation issue as the two offending config options are invalid for the `virtualbox-vm` builder. They are common to the iso and ovf builders which is why they show up in the common generated docs but they should be removed for this builder. 

Is this the first time you’ve run this build? Or was it working with a previous 1.5.x version of Packer? 

With that said, if you remove the two options and leave `”guest_additions_mode”: “attach”` the build should proceed. The alternative is to use the default option and not set `guest_additions_mode` which would upload the guest additions to the guest vm.

**Comment by harisund** [comment]
Ok I kinda guessed as much, since I read in other issues that the documentation doesn't always "keep pace with" the code, so to speak. 

I completely understand "guest_os_type" not being a part of the VM builder, since the user has already created the VM. However, how do I specify the attach option for guest additions? I don't want to upload, I want to attach the guest additions ISO to the VM. Is there no way of doing this at all now? 

(I believe I have used the guest additions mode option at least with virtualbox-vm before. Is the removal a new option?)

**Comment by nywilken** [comment]
Hi @harisund apologies for the confusion, the two options I was referring to are `guest_os_type` and `guest_additions_interface`. If you specify `guest_additions_mode` as "attach" the virtualbox-vm builder, as it is currently written, will use the the IDE interface. 

It looks like a simple enough code change to support `guest_additions_interface`. But let me figure out if there was a reason why it wasn't added in the first place before suggesting a code change.

**Comment by harisund** [comment]
No, I just realized I had some misunderstanding as well, which is why I edited my post. 

Thank you for the clarification. For what it's worth, I don't mind attaching as IDE, but was just curious nevertheless. Attaching is what's important to me, so glad that still works.

Feel free to close this thread if you are tracking elsewhere that the documentation https://www.packer.io/docs/builders/virtualbox-vm needs to be updated. 

Thank you!
